block/internal/syncing/syncer.go

```diff
@@ -471,6 +463,16 @@ func (s *Syncer) trySyncNextBlock(event *common.DAHeightEvent) error {
 		return err
 	}
 
+	// Verify forced inclusion transactions if configured
+	if err := s.verifyForcedInclusionTxs(currentState, data); err != nil {
```

Comment:
The verification fetches the force inclusion TX from DA. This process could be decoupled to speed up the event consumer.